packages/@aws-cdk/aws-servicediscovery/lib/alias-target-instance.ts

```diff
@@ -0,0 +1,74 @@
+import cdk = require('@aws-cdk/cdk');
+import { BaseInstanceProps, InstanceBase } from './instance';
+import { NamespaceType } from './namespace';
+import { DnsRecordType, IService, RoutingPolicy } from './service';
+import { CfnInstance } from './servicediscovery.generated';
+
+/*
+ * Properties for an AliasTargetInstance
+ */
+export interface AliasTargetInstanceProps extends BaseInstanceProps {
+  /**
+   * DNS name of the target
+   */
+  dnsName: string;
+
+  /**
+   * The Cloudmap service this resource is registered to.
+   */
+  service: IService;
+}
+
+/*
+ * Instance that uses Route 53 Alias record type. Currently, the only resource types supported are Elastic Load
+ * Balancers.
+ */
+export class AliasTargetInstance extends InstanceBase {
+  /**
+   * The Id of the instance
+   */
+  public readonly instanceId: string;
+
+  /**
+   * The Cloudmap service to which the instance is registered.
+   */
+  public readonly service: IService;
+
+  /**
+   * The Route53 DNS name of the alias target
+   */
+  public readonly dnsName: string;
+
+  constructor(scope: cdk.Construct, id: string, props: AliasTargetInstanceProps) {
+    super(scope, id);
+
+    if (props.service.namespace.type === NamespaceType.Http) {
```

Comment:
That validation will no longer be sufficient if/when new `NamespaceType` values are added. Would prefer making the test more restrictive (check only for the allowed value).